src/deepsparse/transformers/pipelines/zero_shot_text_classification.py

```diff
@@ -0,0 +1,285 @@
+    default_model_path=(
+        "zoo:nlp/text_classification/distilbert-none/pytorch/huggingface/"
+        "mnli/pruned80_quant-none-vnni"
+    ),
+)
+class ZeroShotTextClassificationPipeline(TransformersPipeline):
+    """
+    transformers zero-shot zero shot text classification pipeline
+
+    example instantiation:
+    ```python
+    zero_shot_text_classifier = Pipeline.create(
+        task="zero_shot_text_classification",
+        model_scheme="nli",
+        model_scheme_config={"hypothesis_template": "This text is related to {}"},
+        model_path="nli_model_dir/",
+    )
+    ```
+
+    example batch size 1, single sequence and three labels:
+    ```python
+    sequence_to_classify = "Who are you voting for in 2020?"
+    candidate_labels = ["Europe", "public health", "politics"]
+    zero_shot_text_classifier(sequence_to_classify, candidate_labels)
+    >>> sequences=['Who are you voting for in 2020?']
+        labels=[['politics', 'Europe', 'public health']]
+        scores=[[0.7635, 0.1357, 0.1007]]
+    ```
+
+    :param model_path: sparsezoo stub to a transformers model, an ONNX file, or
+        (preferred) a directory containing a model.onnx, tokenizer config, and model
+        config. If no tokenizer and/or model config(s) are found, then they will be
+        loaded from huggingface transformers using the `default_model_name` key
+    :param engine_type: inference engine to use. Currently supported values include
+        'deepsparse' and 'onnxruntime'. Default is 'deepsparse'
+    :param batch_size: static batch size to use for inference. Default is 1
+    :param num_cores: number of CPU cores to allocate for inference engine. None
+        specifies all available cores. Default is None
+    :param scheduler: (deepsparse only) kind of scheduler to execute with.
+        Pass None for the default
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param sequence_length: sequence length to compile model and tokenizer for.
+        Default is 128
+    :param default_model_name: huggingface transformers model name to use to
+        load a tokenizer and model config when none are provided in the `model_path`.
+        Default is "bert-base-uncased"
+    :param model_scheme: training scheme used to train the model used for zero shot.
+        Currently supported schemes are "nli"
+    :param model_scheme_config: Config object or a dict of config keyword arguments
+    :param multi_class: True if class probabilities are independent, default False
+    """
+
+    def __init__(
+        self,
+        *,
+        model_scheme: str = ModelSchemes.nli.value,
+        model_scheme_config: Optional[Union[NliTextClassificationConfig, dict]] = None,
+        multi_class: bool = False,
```

Comment:
could `multi_class` be a part of `NliTextClassificationInput`? i know you said before you wanted `hypothesis_template` in the config

Reply:
I made an upstream commit here: https://github.com/neuralmagic/deepsparse/pull/450/commits/05ba598f56f122184863ffbd0995fd55e16900b6

`hypothesis_template` and `multi_class` now exist in both the nli config and nli input. If either are provided in the input, they override the default values in the config.